### ISSUE TITLE
Resolve root_dir and workspace_folders paths before starting LSP client

### ISSUE THREAD

**Issue by chipsenkbeil** [chipsenkbeil/distant.nvim#16]
If the directory is a symlink, we can have a problem where none of the files are triggered because they have the canonicalized file path whereas the root dir (and workspace folder dirs) keep whatever is provided in settings.

Because of that, we need to send a metadata request for each of these directories to set the appropriate path at time of client start.

This also would benefit from supporting batch requests, which starts by implementing https://github.com/chipsenkbeil/distant/issues/20

**Comment by chipsenkbeil** [comment]
Not going to do this now. Given that #86 is available, the user can do this when configuring the `root_dir`.

If we want to use `workspace_folders`, a separate issue could be filed to support async configuration of LSP as a whole.
